test/instrumentation/modules/@aws-sdk/client-sqs.test.js

```diff
@@ -0,0 +1,299 @@
+              },
+            },
+            message: { queue: { name: 'elasticapmtest-queue-1.fifo' } },
+          },
+          outcome: 'success',
+        },
+        'sendMessageBatch',
+      );
+
+      // There will be one or more `SQS ReceiveMessage ...` spans for the ReceiveMessage
+      // API calls until all messages are retrieved -- with interspersed
+      // `SQS DELETE_BATCH ...` spans to delete those messages as they are
+      // received.
+      const spanLinks = [];
+      const pollSpans = [];
+      while (spans.length > 0 && spans[0].name.startsWith('SQS POLL')) {
+        pollSpans.push(spans.shift());
+      }
+
+      pollSpans.forEach((span) => {
+        let numSpanLinks = 0;
+
+        if (span.links) {
+          numSpanLinks = span.links.length;
+          spanLinks.push(...span.links);
+          delete span.links;
+        }
+
+        t.deepEqual(
+          span,
+          {
+            name: 'SQS POLL from elasticapmtest-queue-1.fifo',
+            type: 'messaging',
+            subtype: 'sqs',
+            action: 'poll',
+            context: {
+              service: {
+                target: {
+                  type: 'sqs',
+                  name: 'elasticapmtest-queue-1.fifo',
+                },
+              },
+              destination: {
+                address: LOCALSTACK_HOST,
+                port: 4566,
+                cloud: { region: 'us-east-2' },
+                service: {
+                  type: '',
+                  name: '',
+                  resource: 'sqs/elasticapmtest-queue-1.fifo',
+                },
+              },
+              message: { queue: { name: 'elasticapmtest-queue-1.fifo' } },
+            },
+            outcome: 'success',
+          },
+          `receiveMessage (${numSpanLinks} span links)`,
+        );
+      });
+
+      t.deepEqual(
+        spans.shift(),
+        {
+          name: 'SQS DELETE_BATCH from elasticapmtest-queue-1.fifo',
+          type: 'messaging',
+          subtype: 'sqs',
+          action: 'delete_batch',
+          context: {
+            service: {
+              target: {
+                type: 'sqs',
+                name: 'elasticapmtest-queue-1.fifo',
+              },
+            },
+            destination: {
+              address: LOCALSTACK_HOST,
+              port: 4566,
+              cloud: { region: 'us-east-2' },
+              service: {
+                type: '',
+                name: '',
+                resource: 'sqs/elasticapmtest-queue-1.fifo',
+              },
+            },
+            message: { queue: { name: 'elasticapmtest-queue-1.fifo' } },
+          },
+          outcome: 'success',
+        },
+        'deleteMessageBatch',
+      );
+
+      t.deepEqual(
+        spanLinks,
+        [
+          { trace_id: tx.trace_id, span_id: sendMessageSpanId },
+          // XXX: not sure why this one is not appearing
+          { trace_id: tx.trace_id, span_id: sendMessagesBatchSpanId },
```

Comment:
Note for the reviewer: reason why it's still a draft. I have to check why we do not have the 2nd link. Probably not adding the message attributes properly when sending the batch

Reply:
Comparing to sqs.test.js (for v2), I think you should expect to get *three* span links. An issue with this test (it might not be the reason for this failure) is that it is assuming that the order spans will be:
- all of the ReceiveMessage spans first,
- then a single DeleteMessageBatch span

That's wrong. See the handling in sqs.test.js. There can be one to three DeleteMessageBatch calls interspersed with the ReceiveMessage calls.

Reply:
Got it. I was not handling properly the `Entries` of `SendMessageBatch` command and not propagating the attributes properly. The fix has been pushed :)

Reply:
Great. 

FWIW, I think there is still the potential logic error. The spans might be:
- ReceiveMessage (no messages received)
- ReceiveMessage (1 of the 3 messages received)
- DeleteMessageBatch (to delete that 1 received message)
- RecieveMessage (the remaining 2 messages received)
- DeleteMessageBatch (to delete those last 2 messages)

IIUC the current test code will fail on this case.

Reply:
You're right. I've been lucky the order was the same in my tests. I've changed to check the span nam within the loop like `sqs.test.js` file does.

Reply:
done!